openedx/features/pakx/lms/overrides/tasks.py

```diff
@@ -0,0 +1,127 @@
+        'status': data.get('status')
+    })
+
+    return message_context
+
+
+@task(name='send_reminder_email')
+def send_reminder_email(data):
+    """
+    A task to send reminder emails
+    :param data: (dict) dict containing data for the email
+    """
+
+    site = get_current_site()
+    message_context = _get_email_message_context(data)
+    user = User.objects.filter(username=data.get('username'), email=data.get('email'))
+    user.is_anonymous = False
+    log.info("sending email :{} {} {}".format(site, message_context, user))
+    with emulate_http_request(site, user):
+        msg = CourseProgress().personalize(
+            recipient=Recipient(data.get('username'), data.get('email')),
+            language=data.get('language'),
+            user_context=message_context
+        )
+        ace.send(msg)
+
+
+@task(name='add_enrollment_record')
+def add_enrollment_record(user, course_id):
+    """
+    Add current enrollment record
+    :param user: (settings.User) user object
+    :param course_id: (CourseKeyField) course key
+    """
+
+    try:
+        CourseProgressEmailModel.objects.create(user=user, course_id=course_id)
+    except IntegrityError:
+        log.info("Enrollment record for {} & user:{} already registered".format(course_id, user))
+
+
+@task(name='remove_enrollment_record')
+def remove_enrollment_record(user, course_id):
+    """
+    Remove current enrollment record
+    :param user: (settings.User) user object
+    :param course_id: (CourseKeyField) course key
+    """
+
+    record = CourseProgressEmailModel.objects.filter(user=user, course_id=course_id)
+    if record:
+        record.delete()
+
+
+@task(name='copy_active_course_enrollments')
+def copy_active_course_enrollments():
+    """
+    A task that copies active enrollments to CourseProgressEmailModel model
+    """
+    active_enrollments = CourseEnrollment.objects.filter(is_active=True)
+    log.info("Found {} active enrollment records, ".format(len(active_enrollments)))
+    for enrollment in active_enrollments:
+        add_enrollment_record(enrollment.user, enrollment.course.id)
+
+
+@task(name='send_course_completion_email')
+def check_and_send_email_to_course_learners():
+    """
+    A task that checks all the current progress models & sends course completion or reminder emails
+    """
+    progress_models = CourseProgressEmailModel.objects.filter(status__lt=2)
+    log.info("Sending reminders, found {} active models".format(len(progress_models)))
+    for item in progress_models:
+        course_overview = CourseOverview.get_from_id(item.course_id)
+        grades = CourseGradeFactory().read(user=item.user, course_key=item.course_id)
+        data = {'course_name': course_overview.display_name,
+                'username': item.user.username, 'email': item.user.email,
+                'language': get_user_preference(item.user, LANGUAGE_KEY),
+                'status': grades.percent >= 1.0}
+        if grades.percent >= 1.0:
```

Comment:
What if `grades.percent is 100.00` do we need to send a reminder email in that case as well?